### ISSUE TITLE
Templates rely on transitive dependencies

### ISSUE THREAD

**Issue by dboldureanu** [Alfresco/acs-deployment#716]
When running community with an external PostgreSQL database, getting:

`Error: INSTALLATION FAILED: template: alfresco-content-services/templates/ingress-share.yaml:3:24: executing "alfresco-content-services/templates/ingress-share.yaml" at <{{template "common.capabilities.ingress.apiVersion" .}}>: template "common.capabilities.ingress.apiVersion" not defined`

**Reason:**
- `common.capabilities.ingress.apiVersion` is declared in `https://artifacthub.io/packages/helm/bitnami/common`
- `https://artifacthub.io/packages/helm/bitnami/common` comes as a transient dependency from `https://artifacthub.io/packages/helm/bitnami/postgresql`
- when `postgresql.enabled: false`, the transient dependency will be missing, thus the error.

**Possible solution**:
Add into `helm/alfresco-content-services/charts/alfresco-content-common/Chart.yaml`:
```
dependencies:
  - name: common
    alias: bitnami-common
    version: 1.x.x
    repository: https://charts.bitnami.com/bitnami
```

**Note1**:
I could not add this dependency directly into `helm/alfresco-content-services/Chart.yaml`, because of the of the issue/bug:
[Helm does not support having two dependencies using the same name but from a different repository](https://github.com/helm/helm/issues/7413)
`Chart.yaml` already contains one `common` dependency:
```
  - name: common
    alias: alfresco-digital-workspace
    repository: https://activiti.github.io/activiti-cloud-helm-charts
    version: 7.3.0
    condition: alfresco-content-services.alfresco-digital-workspace.enabled,alfresco-digital-workspace.enabled
```

**Note2**:
In order for dependency to be accessible, I had to call `helm dependency update` inside `helm/alfresco-content-services` folder.
Thus my solution is far from being perfect...
I would appreciate if someone could suggest something better.

**Comment by mteodori** [comment]
can you specify which version of the chart and full command line and values you're passing? the bitnami chart is not used only in postgresql

**Comment by dboldureanu** [comment]
> can you specify which version of the chart and full command line and values you're passing? the bitnami chart is not used only in postgresql

**Versions**
**5.2.0** and **master**

**values.yaml**
```
global:
  tracking:
    auth: secret
    sharedsecret: 33m333cret  

repository:
  edition: Community
  environment:
    JAVA_OPTS: ' -Dsolr.base.url=/solr -Dindex.subsystem.name=noindex -Dsolr.sharedSecret=justAPlaceholderNotUsed -Dalfresco.cluster.enabled=true
      -Ddeployment.method=HELM_CHART -Dtransform.service.enabled=true -XX:MinRAMPercentage=50
      -XX:MaxRAMPercentage=80 -Dencryption.keystore.type=JCEKS -Dencryption.cipherAlgorithm=DESede/CBC/PKCS5Padding
      -Dencryption.keyAlgorithm=DESede -Dencryption.keystore.location=/usr/local/tomcat/shared/classes/alfresco/extension/keystore/keystore
      -Dmetadata-keystore.aliases=metadata -Dmetadata-keystore.metadata.algorithm=DESede '    
  replicaCount: 1
  image:
    repository: alfresco/alfresco-content-repository-community
    tag: 7.2.0
  resources:            
    limits:            
      memory: 3000Mi
    requests:            
      memory: 3000Mi
    
database:            
  driver: org.postgresql.Driver            
  external: true            
  password: 'password'            
  url: jdbc:postgresql://mydb.postgres.database.azure.com:5432/$(CUSTOMERNAME)            
  user: cleopatra
  
activemq:
  persistence:
    existingClaim: $(CUSTOMERNAME)-azurefile1
  replicaCount: 1
  resources:
    limits:
      memory: 1024Mi
    requests:
      memory: 512Mi
  
persistance:            
  enabled: true            
  storageclass:            
    name: jz-azurefile
    
persistence:
  existingClaim: $(CUSTOMERNAME)-azurefile1
  
externalHost: $(CUSTOMERNAME).acs.cleopatraenterprise.eu            
externalPort: 80            
externalProtocol: http

share:
  replicaCount: 1
  image:
    repository: alfresco/alfresco-share
    tag: 7.2.0.2
  resources:
    limits:
      memory: 1280Mi
    requests:
      memory: 1280Mi
    
alfresco-search:
  enabled: false
  searchServicesImage:
    tag: 2.0.3
  searchServicesImage:
    repository: alfresco/alfresco-search-services
  
postgresql:
  enabled: false  
    
pdfrenderer:
  image:
    tag: 2.5.7
  replicaCount: 1
 
imagemagick:
  image:
    tag: 2.5.7
  replicaCount: 1
  
libreoffice:
  image:
    tag: 2.5.7
  replicaCount: 1
  
tika:
  image:
    tag: 2.5.7
  replicaCount: 1
  
transformmisc:
  image:
    tag: 2.5.7
  replicaCount: 1
  
aiTransformer:
  image:
    tag: 1.4.2
  replicaCount: 1
  
transformrouter:
  image:
    tag: 1.5.3
  replicaCount: 1

filestore:
  image:
    tag: 0.16.1
  
# Disable features
alfresco-digital-workspace:
  enabled: false
alfresco-sync-service:
  syncservice:
    enabled : false
    image:
      tag: 4.0.0-M1
postgresql-syncservice:
  enabled: false
ai:
  enabled: false
s3connector:
  enabled: false
email:
  server:
    enabled: false
  inbound:
    enabled: false
imap:
  server:
    enabled: false
```

**Comment by scila1996** [comment]
I have same issue, and I was discovery source code, not found anything line to define "common.capabilities.ingress.apiVersion". But all resource always render this it !

**Comment by scila1996** [comment]
In Chart.yaml ?

https://github.com/Alfresco/acs-deployment/blob/588cc8fc8a4469aa7bbe3e5ebd55056a05083b09/helm/alfresco-content-services/Chart.yaml#L28

Common dependencies have something wrong :)

```
  - name: common
    alias: alfresco-digital-workspace
    repository: https://activiti.github.io/activiti-cloud-helm-charts
    version: 7.3.0
    condition: alfresco-content-services.alfresco-digital-workspace.enabled,alfresco-digital-workspace.enabled
```

**Comment by alxgomz** [comment]
We believe this issue has been fixed with the introduction of the `alfresco-common` library chart.
It's available starting with 5.4.0-M1
